ext/re2/extconf.rb

```diff
@@ -9,370 +9,278 @@
 
-      LIBS
-          Add this string to the flags passed to the linker
-HELP
+          Environment variables used:
 
-#
-#  utility functions
-#
-def config_system_libraries?
-  enable_config("system-libraries", ENV.key?('RE2_USE_SYSTEM_LIBRARIES'))
-end
+            CC
+                Use this path to invoke the compiler instead of `RbConfig::CONFIG['CC']`
 
-def config_cross_build?
-  enable_config("cross-build")
-end
+            CPPFLAGS
+                If this string is accepted by the C preprocessor, add it to the flags passed to the C preprocessor
 
-def concat_flags(*args)
-  args.compact.join(" ")
-end
+            CFLAGS
+                If this string is accepted by the compiler, add it to the flags passed to the compiler
 
-def do_help
-  print(RE2_HELP_MESSAGE)
-  exit!(0)
-end
+            LDFLAGS
+                If this string is accepted by the linker, add it to the flags passed to the linker
 
-def darwin?
-  RbConfig::CONFIG["target_os"].include?("darwin")
-end
+            LIBS
+                Add this string to the flags passed to the linker
+      TEXT
+    end
 
-def windows?
-  RbConfig::CONFIG["target_os"].match?(/mingw|mswin/)
-end
+    private
 
-def freebsd?
-  RbConfig::CONFIG["target_os"].include?("freebsd")
-end
+    def configure_cross_compiler
+      RbConfig::CONFIG["CC"] = RbConfig::MAKEFILE_CONFIG["CC"] = ENV["CC"] if ENV["CC"]
+      RbConfig::CONFIG["CXX"] = RbConfig::MAKEFILE_CONFIG["CXX"] = ENV["CXX"] if ENV["CXX"]
+    end
 
-def target_host
-  # We use 'host' to set compiler prefix for cross-compiling. Prefer host_alias over host. And
-  # prefer i686 (what external dev tools use) to i386 (what ruby's configure.ac emits).
-  host = RbConfig::CONFIG["host_alias"].empty? ? RbConfig::CONFIG["host"] : RbConfig::CONFIG["host_alias"]
-  host.gsub(/i386/, "i686")
-end
+    def build_with_system_libraries
+      header_dirs = [
+        "/usr/local/include",
+        "/opt/homebrew/include",
+        "/usr/include"
+      ]
 
-def target_arch
-  RbConfig::CONFIG['arch']
-end
+      lib_dirs = [
+        "/usr/local/lib",
+        "/opt/homebrew/lib",
+        "/usr/lib"
+      ]
 
-def with_temp_dir
-  Dir.mktmpdir do |temp_dir|
-    Dir.chdir(temp_dir) do
-      yield
+      dir_config("re2", header_dirs, lib_dirs)
     end
-  end
-end
 
-#
-#  main
-#
-do_help if arg_config('--help')
-
-if ENV["CC"]
-  RbConfig::MAKEFILE_CONFIG["CC"] = ENV["CC"]
-  RbConfig::CONFIG["CC"] = ENV["CC"]
-end
-
-if ENV["CXX"]
-  RbConfig::MAKEFILE_CONFIG["CXX"] = ENV["CXX"]
-  RbConfig::CONFIG["CXX"] = ENV["CXX"]
-end
+    def build_with_vendored_libraries
+      message "Building re2 using packaged libraries.\n"
 
-def build_extension(static_p = false)
-  # Enable optional warnings but disable deprecated register warning for Ruby 2.6 support
-  $CFLAGS << " -Wall -Wextra -funroll-loops"
-  $CPPFLAGS << " -Wno-register"
+      abseil_recipe, re2_recipe = load_recipes
```

Comment:
We could take another leaf from @flavorjones’ work on sqlite3-ruby and inline these into this file rather than loading them from a separate `recipes.rb` if we add a flag so we can run `ruby ext/re2/extconf.rb --download-dependencies` to download dependencies rather than our `prepare` Rake task. That doesn’t block this though and can be a separate change.